.circleci/config.yml

```diff
@@ -845,161 +845,202 @@ jobs:
+              only:
                 - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-      - terraform:
+                - master
+      - docker-ci-build:
+          context: docker
+          base-image: circleci
           filters:
             branches:
-              ignore:
+              only:
+                - /^test-[\d|.]+$/
                 - master
+
+  ######################################################################################################################
+  # Will publish release images to dockerhub and aws when commits land on a release/testing branch.
+  ######################################################################################################################
+  release-dockerhub-publish:
+    jobs:
+      - docker-publish:
+          context: docker
+          filters:
+            branches:
+              only:
                 - /^test-[\d|.]+$/
                 - /^release-[\d|.]+$/
-      - prefetch-crates
+
+  ######################################################################################################################
+  # Updates documentation when code is committed to master, and checks for breaking changes.
+  ######################################################################################################################
+  document-publish-workflow:
+    jobs:
+      - prefetch-crates:
+          filters:
+            branches:
+              only: master
       - lint:
           requires:
             - prefetch-crates
+      - build-docs:
+          requires:
+            - lint
+      - deploy-docs:
+          requires:
+            - build-docs
+      - check-breaking-change:
+          requires:
+            - prefetch-crates
+
+  ######################################################################################################################
+  # Used by bors to test prs on master's head before committing to master.
+  # Must support auto and canary branchs...
+  ######################################################################################################################
+  commit-workflow:
+    jobs:
+      - docker-pre-publish:
+          context: docker
           filters:
             branches:
-              ignore:
-                - gh-pages
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
+              only:
+                - auto
+                - canary
+      - lint-docker:
+          filters:
+            branches:
+              only:
+                - auto
+                - canary
+      - terraform:
+          filters:
+            branches:
+              only:
+                - auto
+                - canary
+      - prefetch-crates:
+          filters:
+            branches:
+              only:
+                - auto
+                - canary
+      - lint:
+          requires:
+            - prefetch-crates
       - build-dev:
           requires:
             - prefetch-crates
```

Comment:
Similarly, crious why we don't need to the branch filter here.

Reply:
the requires acts as the filter.... since prefetch crates had a filter, the linter doesn't need it.

Reply:
TIL - that's cool feature.